packages/astro/src/@types/astro.ts

```diff
@@ -2164,6 +2164,35 @@ export interface AstroUserConfig {
 			 * ```
 			 */
 			schema?: EnvSchema;
+
+			/**
+			 * @docs
+			 * @name experimental.env.validateSecretsOnStart
+			 * @kind h4
+			 * @type {boolean}
+			 * @default `false`
+			 * @version 4.11.2
+			 * @description
+			 *
+			 * If enabled, secrets will also be checked on start (dev/build modes). This is useful for example in some CIs to make sure all your secrets are correctly set before deploying.
```

Comment:
The description implies that these secrets aren't validated at the beginning. Maybe we should mention what's the default behaviour of `astro:env`, because - for now - this is our only documentation.